docs/newFeature.md

```diff
@@ -0,0 +1,240 @@
+
+The script needs to implement the `OnNewMessage` and `OnAIResponse` functions. The `OnNewMessage` function is called when a new message is received from the user (chat widget). The `OnAIResponse` function is called when the response from the AI agent is received. After each `OnNewMessage` function call the `OnFeatureResponse` EBus call should be made to notify the UI that the next message can be sent.
+
+```python
+import json
+import argparse
+import azlmbr.bus as bus
+from azlmbr.ai import (
+    ConversationNotificationBus,
+    ConversationNotificationBusHandler,
+    AIAgentNotificationBusHandler,
+    AIAgentRequestBus,
+)
+
+parser = argparse.ArgumentParser(
+    description="Example of creating a Python feature for the GenAIFramework Gem."
+)
+parser.add_argument("agent_id", help="The agent id")
+parser.add_argument("conversation_id", help="The conversation id")
+args = parser.parse_args()
+
+agentId = int(args.agent_id)
+conversationId = int(args.conversation_id)
+
+if "myFeatures" not in globals():
+    myFeatures = []
+
+class MyFeature:
+    def __init__(self, agentId, conversationId):
+        self.agentId = agentId
+        self.conversationId = conversationId
+
+        # Create and connect the Conversation handler and add the OnNewMessage callback.
+        self.handlerConversation = ConversationNotificationBusHandler()
+        self.handlerConversation.connect(self.connectionId)
+        self.handlerConversation.add_callback("OnNewMessage", self.OnNewMessage)
+
+        # Create and connect the Agent handler and add the OnAIResponse callback.
+        self.handlerAgent = AIAgentNotificationBusHandler()
+        self.handlerAgent.connect(self.connectionId)
+        self.handlerAgent.add_callback("OnAIResponse", self.OnAIResponse)
+
+    # This function is called when a new message is received from the user (chat widget).
+    def OnNewMessage(self, message):
+        # The message is a list of length, where the first element is the message from the user.
+        prompt = [{"role": "user", "content": [message[0]]}]
+
+        # A call to the agent can be made to get the history of the conversation.
+        # The history is a json formatted string consisting of a list of messages with roles and content.
+        history = AIAgentRequestBus(bus.Event, "GetHistory", self.agentId)
+        jsonHistory = json.loads(history)
+
+        # The prompt can be modified and sent to the agent. The format of the prompt is a list of messages with roles and content.
+        fullJsonStringPrompt = "..."
+        AIAgentRequestBus(bus.Event, "SendPrompt", self.agentId, fullJsonStringPrompt)
+
+    # This function is called when the response from the AI agent is received.
+    def OnAIResponse(self, message):
+        # The message is a list of length 1, where the first element is the response from the AI agent.
+        jsonMessage = json.loads(message[0])
+
+        response = "..."
+        summary = ["...", "...", ...]
```

Comment:
The naming is not clear in the documentation; the bus expects a _summary_ (short message that is displayed to the user) and a list of detailed messages (a vector of string messages) that are displayed only when requested.